presto-main/src/main/java/com/facebook/presto/operator/scalar/MapEmptyConstructor.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.scalar;
+
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.block.BlockBuilderStatus;
+import com.facebook.presto.spi.block.InterleavedBlockBuilder;
+import com.facebook.presto.spi.function.Description;
+import com.facebook.presto.spi.function.ScalarFunction;
+import com.facebook.presto.spi.function.SqlType;
+import com.google.common.collect.ImmutableList;
+
+import static com.facebook.presto.type.UnknownType.UNKNOWN;
+
+@ScalarFunction("map")
```

Comment:
Put these annotations on the function, since there's only one implementation. You would do it this way on the class if the class implements multiple, specialized versions of the same function. Example here: https://prestodb.io/docs/current/develop/functions.html

You can also remove the name from `@ScalarFunction` since it matches the name of the method (the Java method name is converted from camel case to snake case ... lots of examples in `StringFunctions`)